### PR TITLE
Rebuild docker images

### PR DESCRIPTION
**Pull request by jpienaar**

### CODE REVIEW

**Review by ScottTodd — APPROVED** [review]

File: build_tools/bazel/iree.bazelrc [hunk 1]
```diff
@@ -277,7 +277,7 @@ build:remote_cache_bazel_ci --config=_remote_cache_base
 # specific docker container the CI Bazel builds are run in. The image URL is
 # included for clarity and so that this reference is automatically updated by
 # manage_images.py
-build:remote_cache_bazel_ci --host_platform_remote_properties_override='properties:{name:"cache-silo-key" value:"gcr.io/iree-oss/base-bleeding-edge@sha256:c5f28883e6c570c20128fb37d7af3a00a25df3ce4e2b3a24c3a8dcd183182a27"}'
+build:remote_cache_bazel_ci --host_platform_remote_properties_override='properties:{name:"cache-silo-key" value:"gcr.io/iree-oss/base-bleeding-edge@sha256:cf2e78194e64fd0166f4141317366261d7a62432b72e9a324cb8c2ff4e1a515a"}'
```
Comment (ScottTodd) [inline build_tools/bazel/iree.bazelrc:283]: Bazel build errors 🤔 https://github.com/iree-org/iree/actions/runs/9370945221/job/25799102320?pr=17568#step:4:207
```
ERROR: /work/runtime/src/iree/hal/drivers/vulkan/BUILD.bazel:90:24: Compiling runtime/src/iree/hal/drivers/vulkan/dynamic_symbols.cc failed: (Exit 1): clang failed: error executing command (from target //runtime/src/iree/hal/drivers/vulkan:dynamic_symbols) /usr/lib/llvm-19/bin/clang -U_FORTIFY_SOURCE -fstack-protector -Wall -Wthread-safety -Wself-assign -Wunused-but-set-parameter -Wno-free-nonheap-object -fcolor-diagnostics -fno-omit-frame-pointer -g0 ... (remaining 95 arguments skipped)

Use --sandbox_debug to see verbose messages from the sandbox and retain the sandbox build root for debugging
runtime/src/iree/hal/drivers/vulkan/dynamic_symbols.cc:148:3: error: passing no argument for the '...' parameter of a variadic macro is a C++20 extension [-Werror,-Wc++20-extensions]
  148 |   IREE_RETURN_IF_ERROR(ResolveFunctions(syms.get(), get_proc_addr));
      |   ^
runtime/src/iree/base/status.h:370:3: note: expanded from macro 'IREE_RETURN_IF_ERROR'
  370 |   IREE_STATUS_IMPL_RETURN_IF_API_ERROR_(                \
```

Comment (jpienaar) [reply build_tools/bazel/iree.bazelrc:283]: Mmm, could it be an actual C++20 feature and just never flagged before? That's curious. Haven't checked yet.

Comment (ScottTodd) [reply build_tools/bazel/iree.bazelrc:283]: Yeah idk, we should still be compiling in c++17 mode: https://github.com/iree-org/iree/blob/aef06ed322c2d7789fe1863ff4c884dce6cf03b2/build_tools/bazel/iree.bazelrc#L68-L74

`--config=generic_clang` is on the command: https://github.com/iree-org/iree/actions/runs/9370945221/job/25799102320?pr=17568#step:4:174

Comment (jpienaar) [reply build_tools/bazel/iree.bazelrc:283]: I just don't see this error ... meaning, an argument is passed in . I see https://github.com/iree-org/iree/issues/16946 so perhaps the solution is just to disable that same check for now.

Comment (jpienaar) [reply build_tools/bazel/iree.bazelrc:283]: A trick like https://stackoverflow.com/questions/11761703/overloading-macro-on-number-of-arguments could be used perhaps. Or we just disable the check given things are working and the world moves to C++20 "soon" :)

File: build_tools/bazel/iree.bazelrc [hunk 2]
```diff
@@ -90,6 +90,9 @@ build:generic_clang --per_file_copt=external/.*@-w
 # Would need to then make all the --copt below duplicated with --host_copt.
 build:generic_clang --host_copt=-w
 
+# This is currently flagging the status macros (https://github.com/iree-org/iree/issues/16946)
+build:generic_clang --copt=-Wno-c++20-extensions --host_copt=-Wno-c++20-extensions
```
Comment (ScottTodd) [inline build_tools/bazel/iree.bazelrc:94]: also remove from lower in the file
```
# Explicitly enable some additional warnings.
# Some of these aren't on by default, or under -Wall, or are subsets of warnings
# turned off above.
build:generic_clang --copt=-Wc++20-extensions
```

**Comment by jpienaar** [comment]
This has tests failing on Windows now.

**Comment by ScottTodd** [comment]
> This has tests failing on Windows now.

You can ignore those. https://github.com/actions/runner-images rolled out a bad image and isn't reverting for some reason

**Comment by ScottTodd** [comment]
We don't use Docker or Bazel on Windows, so no way this could have affected those builds, working around the failures in https://github.com/iree-org/iree/pull/17595

**Comment by jpienaar** [comment]
Sorry missed response, SG to submit then.

**Comment by ScottTodd** [comment]
FYI, this appears to have regressed some VMVX benchmarks: https://perf.iree.dev/serie?IREE?007d0bd606ea265d9639e4ec12d65e427dfdbc2c564e72cdd5e3c78a6fa198a6 (5800ms -> 7900ms)

I'm assuming that came from this commit and not some other environment change (e.g. to the runner machines). Maybe due to updating clang or some other toolchain? VMVX may be more sensitive to host compiler optimizations.

**Pull request merged**